### PR TITLE
Add ShortcutKeysEditor tests and fix some bugs

### LINKED ISSUE #1296: ShortcutKeysEditor.EditValue throws InvalidCastException if `value` is not a Shortcut
Should handle this gracefully as if it were null

```cs
public static IEnumerable<object[]> EditValue_TestData()
{
    yield return new object[] { null };
    yield return new object[] { "value" };
    yield return new object[] { Shortcut.CtrlA };
    yield return new object[] { new object() };
}

[Theory]
[MemberData(nameof(EditValue_TestData))]
public void ShortcutKeysEditor_EditValue_ValidProvider_ReturnsValue(object value)
{
    var editor = new ShortcutKeysEditor();
    var mockEditorService = new Mock<IWindowsFormsEditorService>(MockBehavior.Strict);
    var mockServiceProvider = new Mock<IServiceProvider>(MockBehavior.Strict);
    mockServiceProvider
        .Setup(p => p.GetService(typeof(IWindowsFormsEditorService)))
        .Returns(mockEditorService.Object)
        .Verifiable();
    mockEditorService
        .Setup(e => e.DropDownControl(It.IsAny<Control>()))
        .Verifiable();
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Once());
    mockEditorService.Verify(e => e.DropDownControl(It.IsAny<Control>()), Times.Once());

    // Edit again.
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
}
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #1296
Contributes to #817

### CODE REVIEW

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs [hunk 1]
```diff
@@ -12,63 +11,61 @@
...
-        private ShortcutKeysUI shortcutKeysUI;
+        private ShortcutKeysUI _shortcutKeysUI;
 
         /// <summary>
-        ///     Edits the given object value using the editor style provided by ShortcutKeysEditor.GetEditStyle.
+        /// Edits the given object value using the editor style provided by ShortcutKeysEditor.GetEditStyle.
         /// </summary>
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
             if (provider != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs]: Please invert `if` to reduce nesting

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs [hunk 2]
```diff
@@ -12,63 +11,61 @@
...
             {
-                IWindowsFormsEditorService edSvc =
-                    (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
-                if (edSvc != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs]: ditto, and the next line can be reverted

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs [hunk 3]
```diff
@@ -12,63 +11,61 @@
...
+                if (provider.GetService(typeof(IWindowsFormsEditorService)) is IWindowsFormsEditorService edSvc)
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs]: This is the contribution to #817 @russkie

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs [hunk 4]
```diff
@@ -156,13 +153,13 @@ private TypeConverter KeysConverter
...
             /// <summary>
-            ///     Returns the selected keys. If only modifers were selected, we return Keys.None.
+            /// Returns the selected keys. If only modifers were selected, we return Keys.None.
             /// </summary>
             public object Value
             {
                 get
                 {
-                    if (((Keys)currentValue & Keys.KeyCode) == 0)
+                    if (currentValue is Keys currentKeys && (currentKeys & Keys.KeyCode) == 0)
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs]: This is the fix to #1296 @russkie

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs [hunk 5]
```diff
@@ -431,17 +428,16 @@ protected override bool ProcessDialogKey(Keys keyData)
...
             public void Start(IWindowsFormsEditorService edSvc, object value)
             {
                 Debug.Assert(edSvc != null);
-                Debug.Assert(value is Keys);
                 Debug.Assert(!updateCurrentValue);
                 EditorService = edSvc;
                 originalValue = currentValue = value;
 
-                Keys keys = (Keys)value;
+                Keys keys = value is Keys ? (Keys)value : Keys.None;
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/ShortcutKeysEditor.cs]: This is the fix to #1296 @russkie

**Comment by hughbe** [comment]
> Can you please clarify what the bug was?

If you called `new ShortcutKeysEditor().EditValue(new object())`

**Pull request merged**